lstchain/reco/utils.py

```diff
@@ -520,31 +520,40 @@ def filter_events(events,
                   finite_params=None,
                   ):
     """
-    Apply data filtering to a pandas dataframe.
+    Apply data filtering to a pandas dataframe or astropy QTable.
+    The QTable object will be converted to pandas dataframe and used.
     Each filtering range is applied if the column name exists in the DataFrame so that
     `(events >= range[0]) & (events <= range[1])`
+    Returning filter is converted to a numpy object so that it can be used by both dataframe
+    and qtable inputs
 
     Parameters
     ----------
-    events: `pandas.DataFrame`
+    events: `pandas.DataFrame` or 'astropy.table.QTable'
     filters: dict containing events features names and their filtering range
     finite_params: optional, None or list of strings
         extra filter to ensure finite parameters
 
     Returns
     -------
-    `pandas.DataFrame`
+    `pandas.DataFrame` or 'astropy.table.QTable'
     """
+    from astropy.table import QTable
 
-    filter = np.ones(len(events), dtype=bool)
+    if isinstance(events, QTable):
```

Comment:
Maybe just use `Table`? Then it will work for both `Table` and `QTable`, since:

```

In [3]: issubclass(QTable, Table)
Out[3]: True
```

Reply:
Ok